src/routes/targeted-messaging/targeted-messaging.service.ts

```diff
@@ -44,27 +45,54 @@ export class TargetedMessagingService {
     }
   }
 
+  private async getOrCreateTargetedSafe(args: {
+    outreachId: number;
+    safeAddress: `0x${string}`;
+  }): Promise<TargetedSafe> {
+    try {
+      return await this.repository.getTargetedSafe(args);
+    } catch (err) {
+      const outreach = await this.repository.getOutreach({
+        outreachId: args.outreachId,
+      });
+
+      if (!outreach.targetAll) {
```

Comment:
It seems like if the outreach does not exist we get an error trying to access `targetAll`. If it's intentional I would call the method `getOutreachOrFail` to make it clear that it throws if the entity is not found.

Reply:
This was intended yeah, so if the requested outreach doesn't exist you get this response:
`{  "message": "Outreach not found",  "error": "Unprocessable Entity",  "statusCode": 422}`
So I'll change to getOutreachOrFail

Reply:
Nit: this ties in with my aforementioned usage of `NotFoundException` - `422` is validation-specific.

Reply:
Use NotFoundException: https://github.com/safe-global/safe-client-gateway/pull/2194/commits/65896bde7d8fef30ed73e817916f179ddf6cf334#diff-7b46005236ccadec6f3c236cb143960dcc7f8c53afb914e828c907dd112384dbL118

Name change: https://github.com/safe-global/safe-client-gateway/pull/2194/commits/65896bde7d8fef30ed73e817916f179ddf6cf334#diff-c00fbe12e870d92628cb543f6ad60251cca6ee18294c7a674612797d9e02cb1fL55